src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs

```diff
@@ -816,40 +817,33 @@ private void InitializeFSM()
                         .Select(m => m.UniqueAddress)
                         .FirstOrDefault();
 
-                    if (senderUniqueAddress == null)
-                    {
-                        // from unknown node, ignore
-                        Log.Info("Ignoring TakeOver request from unknown node in BecomingOldest from [{0}]", senderAddress);
-                    }
-                    else
+                    switch (senderUniqueAddress)
                     {
-                        if (becomingOldest.PreviousOldest != null)
-                        {
-                            if (becomingOldest.PreviousOldest.Equals(senderUniqueAddress))
-                            {
-                                Sender.Tell(HandOverToMe.Instance);
-                            }
-                            else
+                        case null:
+                            // from unknown node, ignore
+                            Log.Info("Ignoring TakeOver request from unknown node in BecomingOldest from [{0}]", senderAddress);
+                            return Stay();
```

Comment:
@alexvaluyskiy semantically the same - returning `null` returns `stay()` by default.